### PR TITLE
fix: can't rewatch a directory that has been deleted once before

### PR DESCRIPTION
**Pull request by leihaojun**
Fixes https://github.com/sxyazi/yazi/issues/1313

The retrun is_ok from to_unwatch a `deleted path` is always false. 
image

But the directory has actually been unwatch, because the return is false, so it was not actively removed from the `WATCHED`, 
image

so when the same name path is created, the path recognize as a watching path in  `old` var.  
image

so the new path with the same name will not add to the `to_watch` to rewatcher.

### CODE REVIEW

**Comment by sxyazi** [comment]
Thanks for the PR!

File: yazi-core/src/manager/watcher.rs [hunk 1]
```diff
@@ -72,7 +72,7 @@ impl Watcher {
 				(old.difference(new).cloned().collect(), new.difference(old).cloned().collect())
 			};
 
-			to_unwatch.retain(|u| watcher.unwatch(u).is_ok());
+			to_unwatch.retain(|u| watcher.unwatch(u).is_ok() || !u.exists());
```
Comment (sxyazi) [inline yazi-core/src/manager/watcher.rs:75]: `.unwatch()` should already return an error when it fails to unwatch a non-existent directory, why don't we just use it directly?

Comment (leihaojun) [reply yazi-core/src/manager/watcher.rs:75]: you mean to analyze the cause of the error before deciding whether to remove the path?

Comment (sxyazi) [reply yazi-core/src/manager/watcher.rs:75]: Yep, I'll fix it now

**Review by sxyazi — APPROVED** [review]
Thanks!

**Pull request merged**